assets/javascripts/discourse/initializers/admin-menu-webinar-button.js

```diff
@@ -8,35 +8,60 @@ import WebinarPicker from "../components/modal/webinar-picker";
 const PLUGIN_ID = "discourse-zoom";
 
 function initialize(api) {
-  api.decorateWidget("topic-admin-menu:adminMenuButtons", (helper) => {
-    const topic = helper.attrs.topic;
-    const { canManageTopic } = helper.widget.currentUser || {};
+  if (api.addTopicAdminMenuButton) {
+    api.addTopicAdminMenuButton((topic) => {
+      const canManageTopic = api.getCurrentUser()?.canManageTopic;
 
-    if (!topic.isPrivateMessage && canManageTopic) {
-      return {
-        buttonClass: "btn-default",
-        action: topic.webinar ? "removeWebinar" : "addWebinar",
-        icon: "shield-alt",
-        fullLabel: topic.webinar ? "zoom.remove_webinar" : "zoom.add_webinar",
-      };
-    }
-  });
+      if (!topic.isPrivateMessage && canManageTopic) {
+        return {
+          icon: "shield-alt",
+          label: topic.get("webinar")
+            ? "zoom.remove_webinar"
+            : "zoom.add_webinar",
+          action: () => {
+            if (topic.get("webinar")) {
+              const dialog = api.container.lookup("service:dialog");
+              const topicController = api.container.lookup("controller:topic");
+              removeWebinar(topic, dialog, topicController);
+            } else {
+              const modal = api.container.lookup("service:modal");
+              showWebinarModal(topic, modal);
+            }
+          },
+        };
+      }
+    });
+  } else {
+    api.decorateWidget("topic-admin-menu:adminMenuButtons", (helper) => {
```

Comment:
Maybe add a note here to drop the `else` part of the clause in a month or two. No need to carry this forward forever.